src/cluster.c

```diff
@@ -1935,6 +1935,15 @@ int clusterProcessPacket(clusterLink *link) {
          * of the message type. */
         if (!sender && type == CLUSTERMSG_TYPE_MEET)
             clusterProcessGossipSection(hdr,link);
+	    
+	if (sender && type == CLUSTERMSG_TYPE_PING && link->node == NULL){
+            link->node = sender;
```

Comment:
I'm not sure I understand the failure mode here. The event handler always has the reference to the link, so the assumption is that eventually the kernel will realize the connection is dead and close the connection, which will than cause it to be cleaned up: https://github.com/redis/redis/blob/d0f36eeda013213919afe89bbc8a1d5bb3c94340/src/cluster.c#L2388.

Reply:
Nothing can trigger  handleLinkIOError() because  the partition may cause tcp message lost. Am I right?

Reply:
The inbound link is closed and freed when the underlying connection is closed. I would expect that this is supposed to be solved by tcp-keep alive. (Which I don't think we set on cluster nodes, but maybe we should?) The TCP keep alive will eventually cause the kernel to be aware the connection is closed, it will trigger the read handler, which will free the link.

Reply:
Your way is also OK. I have changed these files and set the keep-alive interval as 2*server.cluster_node_timeout.
In the worst case, B will ping A after server.cluster_node_timeout/2. And if there is a partition,  B can't receive PONG from A after another  server.cluster_node_timeout/2. Then  B will free the older link. The corresponding "struct clusterLink" in A should be free.  I increase the interval time appropriately to make sure that has happened. By default, after 60s, A will free the structure.
What's your opinion?

Reply:
Sounds good to me!